File: android_samples/26.5 Search4_WebServiceRetrofit/app/src/main/java/com/msnegi/websearchretrofit/web/ServiceCallback.java
package com.msnegi.websearchretrofit.web;

import com.msnegi.websearchretrofit.web.APIServices;

public interface ServiceCallback {
    public void  success(APIServices.ServiceType serviceType,Object result, String msg);
    public void  fail(APIServices.ServiceType serviceType, String msg);
    public void  cancel();
}

File: android_samples_java/12.9 ViewStubDemo/app/src/main/java/com/msnegi/viewstubdemo/webservice/ServiceCallback.kt
package com.msnegi.viewstubdemo.webservice

import com.msnegi.viewstubdemo.web.APIServices

public interface ServiceCallback {
    fun success(serviceType: APIServices.ServiceType, result: Any, msg: String)
    fun fail(serviceType: APIServices.ServiceType, msg: String)
    fun cancel()
}
